### ISSUE TITLE
Filter Item events based on Library by UserFilter

### ISSUE THREAD

**Issue by nothing2obvi** [jellyfin/jellyfin-plugin-webhook#74]
Hello! My Jellyfin setup is a little weird because I basically have 4 copies of the same library but each library uses .STRM files that allow permission for that specific user's Google Drive ID. Anyway, if I have 4 users -- Anne, Bob, Christa, and Dan -- each only with access to their own library, when an episode is added to Bob's library, that episode will only work for Bob.

In the Webhooks plugin settings, I have each user filtered only to themselves (i.e. Bob's webhook only has his name checked). All notification types are checked except Generic, Playback Progress, and Task Completed (using Gotify, as well as Notifiarr to get notifications via Discord).

![example](https://user-images.githubusercontent.com/85228035/138541686-d3bfe782-732f-4006-8545-e103bf915969.png)

And in the "Notifications" tab in the left sidebar, for "New content added" I have "All users" checked (as in the image below).
![Screen Shot 2021-10-22 at 10 38 15 PM](https://user-images.githubusercontent.com/85228035/138541073-336c28c4-478b-4b05-883e-e52e54c2ae83.png)

With this current setup, when an episode is added to "Bob's Library," all users -- Anne, Bob, Christa, and Dan -- get a notification.
![whatshappening](https://user-images.githubusercontent.com/85228035/139362743-2abd8984-7517-408e-af36-59c4c1bbac17.png)

However, when an episode is added to "Bob's Library," I actually only want Bob to get a notification because only Bob has access to "Bob's Library." 
![whatidlike](https://user-images.githubusercontent.com/85228035/139362798-a7db5a07-557e-4f31-9b22-bcdafd2e0a43.png)

Is there a way to filter it so that users only get "Item Added" notifications when an item is added to a library that they specifically have access to?

Or have I screwed myself over with how I've set up my server?
Or am I overlooking something?

Thanks for your help!

(EDIT: It seems I had the "User Data Saved" notification type unchecked too. Not sure if relevant.)

**Comment by Cameron-Gregg** [comment]
Yeah Id like to isolate my kids content from the adults collection. Library id or something would be great.

**Comment by Aydanill** [comment]
Yeah hopefully it gets added. There's a lot of use cases and you can try. For example having users be pinged for specific shows getting a new episode with a combination of having a new user with a tag whitelisted.

**Comment by ph1go** [comment]
Agree with the desire for this feature. In a situation with a single server hosting multiple users, each with different libraries, it'd be invaluable to be able to tailor the notifications per library. Unless I'm missing the point of the user filter (with Pushover), users (who are unticked in the user filter) are getting notifications for items added to libraries that they don't have access to...

**Comment by ph1go** [comment]
@crobibero would you mind confirming if the above (users with access to specific libraries only receiving notifications when media is added to those libraries) is possible? I seem to be going around in circles trying to get it to work through the webhook plugin with no luck. It works correctly with other notification types per user (eg `authentication success` and `playback start`) but `item added` just sends to everyone, no matter what I try. Is that because the user filter isn't aware of libraries and which ones each user has access to?

I've got a couple of Sonarr instances and I can easily run the notifications correctly upstream (`on import complete`) but I'd prefer them to be fired after the media is actually added to Jellyfin. 

Not critical but I'll stop wasting my time if it's not possible. :)

**Comment by ph1go** [comment]
Ok, thanks. It would definitely be a nice addition imo, if it's feasible.
